de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/ResourceActivityQueuer.java

```diff
@@ -0,0 +1,85 @@
+package de.fu_berlin.inf.dpp.session.internal;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import de.fu_berlin.inf.dpp.activities.IActivity;
+import de.fu_berlin.inf.dpp.activities.IResourceActivity;
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.session.IActivityQueuer;
+
+/**
+ * This Resource based Queue gets a {@code Set} of {@link SPath}s, which should
+ * be queued and flushes them when the SPath entry was removed by
+ * {@link #disableQueuing(SPath)}.
+ * <p>
+ * To keep it simple, the execution order of queued activities stays unchanged.
+ * </p>
+ * TODO Discard obsolete Editor Events
+ */
+public class ResourceActivityQueuer implements IActivityQueuer {
+
+    private Set<SPath> queueRessources;
```

Comment:
Rename to `missingResources` or `unavailableResources`? `queueResources` seems weird. These are the resources activities are queued for, but `resourcesToQueueActivitiesFor` would be way to long.